### PR TITLE
Fix golint errors in lib/selector

### PR DESCRIPTION
**Pull request by xychu**
Not sure if this is the right time to do the golint checks. :-)

### CODE REVIEW

File: lib/selector/parser/ast.go [hunk 1]
```diff
@@ -15,31 +15,38 @@
...
 type Selector interface {
 	// Evaluate evaluates the selector against the given labels expressed as a concrete map.
 	Evaluate(labels map[string]string) bool
 	// EvaluateLabels evaluates the selector against the given labels expressed as an interface.
 	// This allows for labels that are calculated on the fly.
 	EvaluateLabels(labels Labels) bool
+	// String returns a string that represents this selector.
 	String() string
-	UniqueId() string
+	// String returns the unique ID that represents this selector.
```
Comment (caseydavenport) [inline lib/selector/parser/ast.go]: String -> UniqueID

File: lib/selector/parser/ast.go [hunk 2]
```diff
@@ -103,34 +112,38 @@ func (node LabelEqValueNode) collectFragments(fragments []string) []string {
 	return append(fragments, node.LabelName, " == ", quote, node.Value, quote)
 }
 
+// LabelInSetNode contains a LabelName and a value of StringSet
 type LabelInSetNode struct {
 	LabelName string
 	Value     StringSet
 }
 
+// Evaluate return true if the labels' value in the set of current LabelInSetNode
```
Comment (caseydavenport) [inline lib/selector/parser/ast.go]: typo: return -> returns

File: lib/selector/parser/ast.go [hunk 3]
```diff
@@ -103,34 +112,38 @@ func (node LabelEqValueNode) collectFragments(fragments []string) []string {
...
 	return collectInOpFragments(fragments, node.LabelName, "in", node.Value)
 }
 
+// LabelNotInSetNode contains a LabelName and a value of StringSet
 type LabelNotInSetNode struct {
 	LabelName string
 	Value     StringSet
 }
 
+// Evaluate return true if the labels' value not in the set of current LabelNotInSetNode
```
Comment (caseydavenport) [inline lib/selector/parser/ast.go]: return -> returns

File: lib/selector/parser/ast.go [hunk 4]
```diff
@@ -160,17 +173,19 @@ func collectInOpFragments(fragments []string, labelName, op string, values Strin
 	return fragments
 }
 
+// LabelNeValueNode contains a LabelName and a value of string
 type LabelNeValueNode struct {
 	LabelName string
 	Value     string
 }
 
+// Evaluate return true if the labels' value not equal to the value of current LabelNeValueNode
```
Comment (caseydavenport) [inline lib/selector/parser/ast.go]: return -> returns

File: lib/selector/parser/ast.go [hunk 5]
```diff
@@ -183,26 +198,30 @@ func (node LabelNeValueNode) collectFragments(fragments []string) []string {
 	return append(fragments, node.LabelName, " != ", quote, node.Value, quote)
 }
 
+// HasNode contains a LabelName
 type HasNode struct {
 	LabelName string
 }
 
+// Evaluate return true if the labels has the LabelName of current HasNode
```
Comment (caseydavenport) [inline lib/selector/parser/ast.go]: return -> returns (and a few more through the PR)

File: lib/selector/parser/stringset.go [hunk 6]
```diff
@@ -49,6 +51,7 @@ func (ss StringSet) Contains(s string) bool {
 	return ss[minIdx] == s
 }
 
+// ConvertToStringSetInPlace convert a string slice into StringSet in place
```
Comment (caseydavenport) [inline lib/selector/parser/stringset.go]: convert -> converts

File: lib/selector/tokenizer/tokenizer.go [hunk 7]
```diff
@@ -70,6 +88,7 @@ var (
 	inRegex         = regexp.MustCompile("^" + inExpr)
 )
 
+// Tokenize transform input string to tokens
```
Comment (caseydavenport) [inline lib/selector/tokenizer/tokenizer.go]: transform -> transforms

**Comment by fasaxc** [comment]
@xychu You code cleanups/name fixes look good to me but a lot of the comments are of the form:
```
// DoX does X
func DoX(...)
```
I'd prefer to have no comment in that case since the comment doesn't add very much.  What do you think?

**Comment by xychu** [comment]
@fasaxc sure, will update it later.

**Comment by xychu** [comment]
@fasaxc @caseydavenport plz review again.

**Pull request merged**